packages/matchbox/src/components/Button/Button.js

```diff
@@ -1,140 +1,149 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
+import { deprecate } from '../../helpers/propTypes';
+import styled from 'styled-components';
+import { Box } from '../Box';
 
 import Group from './Group';
-import styles from './Button.module.scss';
-
-class Button extends Component {
-  static displayName = 'Button';
-
-  static Group = Group;
-
-  static propTypes = {
-    color: PropTypes.oneOf(['orange', 'blue', 'navy', 'purple', 'red']),
-    disabled: PropTypes.bool,
-    destructive: PropTypes.bool,
-    flat: PropTypes.bool,
-    outline: PropTypes.bool,
-    size: PropTypes.oneOf(['small', 'large', 'default']),
-    fullWidth: PropTypes.bool,
-    submit: PropTypes.bool,
-    to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-    external: PropTypes.bool,
-    component: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.element
-    ]),
-    children: PropTypes.node
-  }
-
-  static defaultProps = {
-    size: 'default'
-  }
+import { base, visualSize, colorVariant, disabled, fullWidth, group } from './styles';
+
+const StyledButton = styled(Box)`
+  ${base}
+  ${visualSize}
+  ${colorVariant}
+  ${disabled}
+  ${fullWidth}
+`;
+
+// Button.Group is styled here to access a classname reference to StyledButton
+const StyledGroup = styled(Group)`
+  ${group(StyledButton)}
+`;
+
+function Button(props) {
+  const {
+    children,
+
+    // Styles
+    primary, // Deprecate in favor of color
+    color,
+    disabled,
+    destructive, // Deprecate in favor of color
+
+    // Below 3 props to be deprecated for a 'weight' prop
+    plain, // Deprecate in favor of flat
+    flat,
+    outline,
+
+    // Options
+    // Renaming to prevent `width` and `height` pass through
+    // Size is a valid styled-system prop
+    size: buttonSize,
+    fullWidth,
+    submit,
+
+    to,
+    Component, // Deprecate in favor of component
+    component,
+    external,
+    title,
+
+    // Events
+    onClick,
+    onFocus,
+    onBlur,
+
+    className = '',
+    ...rest // TODO remove spreading of unknown props
+  } = props;
+
+  // Polyfills deprecrated 'Component' prop
+  const WrapperComponent = component || Component;
+
+  // Polyfills to be deprecrated 'primary' and 'destructive' prop
+  const buttonColor = primary ? 'blue' : destructive ? 'red' : color;
+
+  // Experimenting with a weight prop to replace outline, plain, and flat in the future
+  const visualWeight = React.useMemo(() => {
+    if (outline) {
+      return 'normal';
+    }
```

Comment:
For some reason the `"outline"` variant is rendering with a gray background in Chrome in responsive mode (so probably on phones it would look like this):

![Screen Shot 2020-01-21 at 12 25 18 PM](https://user-images.githubusercontent.com/3613392/72827685-3d7b6e80-3c49-11ea-824d-01e10a2a7622.png)

Reply:
Oddly enough, the same thing is happening in Safari with no special settings or anything:
![Screen Shot 2020-01-21 at 12 28 53 PM](https://user-images.githubusercontent.com/3613392/72827908-a1059c00-3c49-11ea-96f8-cdc32fbc9482.png)

Reply:
*and* Firefox. Weird stuff.

<img width="119" alt="Screen Shot 2020-01-21 at 12 29 52 PM" src="https://user-images.githubusercontent.com/3613392/72827999-c72b3c00-3c49-11ea-97fd-b50d1ae2d1ed.png">